scripts/conversion_script.py

```diff
@@ -702,38 +705,36 @@ def main():
     finally:
         # Gather JSON & Textual report
         data = gather_json_report()
-
-        if data:
+        if data and not rollback_errors:
             highest_level = find_highest_report_level(actions=data["actions"])
             # Set the first position of the list as being the final status,
             # that's needed because `find_highest_report_level` will sort out
             # the list with the highest priority first.
             output.status = highest_level
 
-            if not output.message:
-                # Generate report message and transform the raw data into
-                # entries for Insights.
-                output.message, output.alert = generate_report_message(highest_level)
+            # At this point we know JSON report exists and no rollback errors occured
+            # we can rewrite possible previous message with more specific one and set alert
+            output.message, output.alert = generate_report_message(highest_level)
 
-                if "successfully" in output.message:
-                    gpg_key_file.keep = True
+            if "successfully" in output.message:
```

Comment:
Can't we use the resulting status from `highest_level` here?

Reply:
I think we can, should have same effect.. however I rather used `output.alert`, it is simply False if it was successful- seemed cleaner.